File: mobile/src/androidTest/java/org/openhab/habdroid/screengrab/ScreengrabTest.java
package org.openhab.habdroid.screengrab;

import androidx.test.espresso.ViewInteraction;
import androidx.test.espresso.contrib.RecyclerViewActions;
import androidx.test.filters.LargeTest;
import androidx.test.runner.AndroidJUnit4;

import org.junit.ClassRule;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.openhab.habdroid.R;
import org.openhab.habdroid.TestWithoutIntro;
import tools.fastlane.screengrab.locale.LocaleTestRule;

import static androidx.test.espresso.Espresso.onView;
import static androidx.test.espresso.Espresso.pressBack;
import static androidx.test.espresso.action.ViewActions.click;
import static androidx.test.espresso.matcher.ViewMatchers.withId;
import static tools.fastlane.screengrab.Screengrab.screenshot;

@LargeTest
@RunWith(AndroidJUnit4.class)
public class ScreengrabTest extends TestWithoutIntro {
    @ClassRule
    public static final LocaleTestRule localeTestRule = new LocaleTestRule();

    @Test
    public void test() {
        ViewInteraction recyclerView = onView(withId(R.id.recyclerview));

        screenshot("main-menu");

        // open first floor => Office
        recyclerView.perform(RecyclerViewActions.actionOnItemAtPosition(0, click()));
        recyclerView.perform(RecyclerViewActions.actionOnItemAtPosition(1, click()));
        screenshot("office");

        pressBack();
        pressBack();

        // open "Outside Temperature"
        recyclerView.perform(RecyclerViewActions.actionOnItemAtPosition(5, click()));
        screenshot("chart");
        pressBack();

        // open widget overview
        recyclerView.perform(RecyclerViewActions.actionOnItemAtPosition(10, click()));
        screenshot("widget-overview");
    }
}

File: mobile/src/androidTest/java/org/openhab/habdroid/screengrab/ScreengrabTest.kt
/*
 * Copyright (c) 2010-2019 Contributors to the openHAB project
 *
 * See the NOTICE file(s) distributed with this work for additional
 * information.
 *
 * This program and the accompanying materials are made available under the
 * terms of the Eclipse Public License 2.0 which is available at
 * http://www.eclipse.org/legal/epl-2.0
 *
 * SPDX-License-Identifier: EPL-2.0
 */

package org.openhab.habdroid.screengrab

import androidx.recyclerview.widget.RecyclerView
import androidx.test.espresso.Espresso.onView
import androidx.test.espresso.Espresso.pressBack
import androidx.test.espresso.action.ViewActions.click
import androidx.test.espresso.contrib.RecyclerViewActions
import androidx.test.espresso.matcher.ViewMatchers.withId
import androidx.test.filters.LargeTest
import androidx.test.runner.AndroidJUnit4
import org.junit.ClassRule
import org.junit.Test
import org.junit.runner.RunWith
import org.openhab.habdroid.R
import org.openhab.habdroid.TestWithoutIntro
import tools.fastlane.screengrab.Screengrab.screenshot
import tools.fastlane.screengrab.locale.LocaleTestRule

@LargeTest
@RunWith(AndroidJUnit4::class)
class ScreengrabTest : TestWithoutIntro() {

    @Test
    fun test() {
        val recyclerView = onView(withId(R.id.recyclerview))

        screenshot("main-menu")

        // open first floor => Office
        recyclerView.perform(RecyclerViewActions.actionOnItemAtPosition<RecyclerView.ViewHolder>(0, click()))
        recyclerView.perform(RecyclerViewActions.actionOnItemAtPosition<RecyclerView.ViewHolder>(1, click()))
        screenshot("office")

        pressBack()
        pressBack()

        // open "Outside Temperature"
        recyclerView.perform(RecyclerViewActions.actionOnItemAtPosition<RecyclerView.ViewHolder>(5, click()))
        screenshot("chart")
        pressBack()

        // open widget overview
        recyclerView.perform(RecyclerViewActions.actionOnItemAtPosition<RecyclerView.ViewHolder>(10, click()))
        screenshot("widget-overview")
    }

    companion object {
        @ClassRule
        @JvmField
        val localeTestRule = LocaleTestRule()
    }
}
